js-foundations.js

```diff
@@ -2,49 +2,155 @@
 // Coding practical questions
 
 // 1. Write a function that takes a number as an argument and decides if the number is evenly divisble by three or not. Use the test cases 15, 0, 385, -7. Expected output: "15 is divisible by three", "0 is divisible by three", "385 is not divisble by three", "-7 is not divisble by three".
-
-
+// Write a function called divisbleBy
+// Take in an argument of a number, 15, 0, 385, -7
+// Uses an if/else statement to check if %3 === 0
+// Returns the correct statement whether or not it's evenly divisible by 
+
+var numOne = 15
+var numTwo = 0
+var numThree = 385
+var numFour = -7
+
+const divisbleBy = (number) => {
+    if (number%3 === 0){
+        return `${number} is evenly divisible by 3.`
+    } else {
+        return `${number} is not evenly divisible by 3.`
+    }
+}
+console.log(divisbleBy(numOne))
+console.log(divisbleBy(numTwo))
+console.log(divisbleBy(numThree))
+console.log(divisbleBy(numFour))
 
 // 2. Create an object called helloMe. Include your firstName, lastName and TWO OTHER properties of you. Add a method to your object that will return a sentence using the object properties to describe you.
-
+// object called helloMe
+// the object will include firstName, lastName, age, numSiblings
+// Use a method to return a sentence using the properties
+var person = {
+    firstName: "Kristen",
+    lastName: "Mulder",
+    age: 27,
+    numSiblings: 2,
+    getData: function(){
+        return "Hi, my name is " + this.firstName +" "+ this.lastName + ". I am " + this.age + " years old and I have "+this.numSiblings+ " siblings." 
+    }
+}
+console.log(person.getData())
 
 
 // ------------------------------------- Consider the variable:
 
 var randomNouns = ["streetlamp", "potato", "teeth", "conclusion", "nephew", "temperature", "database"]
 
 // 3a. Write a function that returns every other item from the array as one string. Expected output: "streetlamp teeth nephew database"
+// write a function calle everyOther 
+// Takes in an array as an argument 
+// Filters every other index by using indexOf formula to check %2 and returns a new array
+// Joins the new array into a string
 
+const everyOther = (array) => {
+    return array.filter(value => {
+        return array.indexOf(value) %2 === 0
+    }).join(" ")
+}
+console.log(everyOther(randomNouns))
 
```

Comment:
Love the pseudo coding! Remember the filter method can give you the index as one of its arguments. Might save you a little bit of code.